eng/pipelines/coreclr/templates/run-superpmi-job.yml

```diff
@@ -0,0 +1,143 @@
+          ${{ framework }}:
+            _Framework: ${{ framework }}
+    steps:
+    - ${{ parameters.steps }}
+  
+    - script: $(PythonScript) $(Build.SourcesDirectory)/src/coreclr/scripts/superpmi-setup.py -source_directory $(Build.SourcesDirectory) -core_root_directory $(Core_Root_Dir) -arch $(archType) -mch_file_tag $(MchFileTag) -libraries_directory $(Core_Root_Dir) -tests_directory $(managedTestArtifactRootFolderPath) -max_size 50 # size in MB
+      displayName: ${{ format('SuperPMI setup ({0})', parameters.osGroup) }}
+
+      # Run superpmi collection in helix
+    - template: /eng/common/templates/steps/superpmi-send-to-helix.yml
+      parameters:
+        HelixSource: '$(HelixSourcePrefix)/$(Build.Repository.Name)/$(Build.SourceBranch)' # sources must start with pr/, official/, prodcon/, or agent/
+        HelixType: 'test/superpmi/$(Kind)/$(_Framework)/$(Architecture)'
+        HelixAccessToken: $(HelixApiAccessToken)
+        HelixTargetQueues: $(Queue)
+        HelixPreCommands: $(HelixPreCommand)
+        Creator: $(Creator)
+        WorkItemTimeout: 4:00 # 4 hours
+        WorkItemDirectory: '$(WorkItemDirectory)'
+        CorrelationPayloadDirectory: '$(CorrelationPayloadDirectory)'
+        ProjectFile: ${{ parameters.projectFile }}
+        BuildConfig: ${{ parameters.buildConfig }}
+        osGroup: ${{ parameters.osGroup }}
+        LibrariesArtifacts: '$(LibrariesArtifacts)'
+        TestsArtifacts: '$(TestsArtifacts)'
+
+    - task: PublishPipelineArtifact@1
+      displayName: Publish SuperPMI collection
+      inputs:
+        targetPath: $(Build.SourcesDirectory)/artifacts/helixresults
+        artifactName: 'SuperPMI_Result_$(osGroup)$(osSubgroup)_$(archType)_$(buildConfig)_${{ parameters.runtimeType }}_${{ parameters.codeGenType }}'
+      continueOnError: true
+      condition: always()
+
+    - script: $(PythonScript) $(Build.SourcesDirectory)/src/coreclr/scripts/superpmi.py merge-mch -pattern $(MchFilesLocation)libraries.pmi*.mch  -output_mch_path $(MchFilesLocation)libraries.pmi.$(MchFileTag).mch
+      displayName: Merge libraries SuperPMI collections
+
+    # TODO: Disable SPMI for P1 tests
+    # - script: $(PythonScript) $(Build.SourcesDirectory)/src/coreclr/scripts/superpmi.py merge-mch -pattern $(MchFilesLocation)tests.pmi*.mch  -output_mch_path $(MchFilesLocation)tests.pmi.$(MchFileTag).mch
+    #   displayName: Merge tests SuperPMI collections
+
+    # For now, we won't upload merged collection as an artifact.
+
+    - script: $(PythonScript) $(Build.SourcesDirectory)/src/coreclr/scripts/superpmi.py upload -arch $(archType) -build_type $(buildConfig) -mch_files $(MchFilesLocation)libraries.pmi.$(MchFileTag).mch -core_root $(Build.SourcesDirectory)/artifacts/bin/coreclr/$(osGroup).x64.$(buildConfigUpper)
+      displayName: Upload SuperPMI libraries collection to Azure Storage
+      env:
+        CLRJIT_AZ_KEY: $(clrjit_key1) # secret key stored as variable in pipeline
```

Comment:
We usually try to avoid key tokens in public pipelines. Is this pipeline public?

Reply:
The pipeline is internal, precisely due to this requirement.